packages/world/src/utils/ReadUtils.sol

```diff
@@ -53,34 +49,34 @@ struct EntityData {
 }
 
 function getEntityInventory(EntityId entityId) view returns (InventoryObject[] memory) {
-  uint16[] memory objectTypeIds = InventoryObjects._get(entityId);
-  InventoryObject[] memory inventoryObjects = new InventoryObject[](objectTypeIds.length);
-  bytes32[] memory allInventorys = ReverseInventoryEntity._get(entityId);
-  for (uint256 i = 0; i < objectTypeIds.length; i++) {
-    ObjectTypeId objectTypeId = ObjectTypeId.wrap(objectTypeIds[i]);
-    uint16 count = InventoryCount._get(entityId, objectTypeId);
-    bool isTool = objectTypeId.isTool();
-    uint256 numEntities = 0;
-    if (isTool) {
-      for (uint256 j = 0; j < allInventorys.length; j++) {
-        if (ObjectType._get(EntityId.wrap(allInventorys[j])) == objectTypeId) {
-          numEntities++;
-        }
-      }
-    }
-    InventoryEntity[] memory inventoryEntities = new InventoryEntity[](numEntities);
-    if (numEntities > 0) {
-      uint256 k = 0;
-      for (uint256 j = 0; j < allInventorys.length; j++) {
-        EntityId tool = EntityId.wrap(allInventorys[j]);
-        if (ObjectType._get(tool) == objectTypeId) {
-          inventoryEntities[k] = InventoryEntity({ entityId: tool, mass: Mass._getMass(tool) });
-          k++;
-        }
-      }
-    }
-    inventoryObjects[i] =
-      InventoryObject({ objectTypeId: objectTypeId, numObjects: count, inventoryEntities: inventoryEntities });
-  }
-  return inventoryObjects;
+  // uint16[] memory objectTypeIds = InventoryObjects._get(entityId);
```

Comment:
oh hmm, with this removed the client wont be able to sync inventory cuz its using the getter for now, so this will break mine and builds

Is there a way to have this getter still work @vdrg (even if its gas expensive)?

Otherwise, @alvrs will the change to indexer to use slots be easy or take a while? wondering if we should merge this in and have broken inventory or wait

Reply:
ah yeah I can just iterate over all the slots and get the types

Reply:
okay sweet yea then at least it will still work even if its expensive